File: tests/fixtures/sample-project/services/user-service/src/models/notification.model.ts
/**
 * Notification models — tests compound type annotations.
 *
 * Tests:
 * - Interface used as property type with union (X | null) — usedAsType FP
 * - Interface used via extends — usedAsType via inheritance
 */

export interface NotificationChannel {
  name: string;
  enabled: boolean;
}

export interface NotificationPriority {
  level: number;
  label: string;
}

export interface BaseNotification {
  title: string;
  body: string;
}

export interface Notification extends BaseNotification {
  channel: NotificationChannel | null;
  priority: NotificationPriority | undefined;
  recipients: string[];
}

File: tests/fixtures/sample-python-project-negative/services/user_service/models/notification_model.py
"""Notification models — tests compound type annotations in class properties.

Tests:
- Class used as property type with Optional[X] and X | None (usedAsType FP)
- Class used as superclass (usedAsType via inheritance)
- Mapped[] with compound types
"""

from dataclasses import dataclass, field
from typing import Optional


@dataclass
class NotificationChannel:
    """Channel config — used as a property type in Notification."""
    name: str
    enabled: bool = True


@dataclass
class NotificationPriority:
    """Priority — used via Optional[NotificationPriority] annotation."""
    level: int
    label: str


@dataclass
class BaseNotification:
    """Base class — used via inheritance (usedAsType via superClass)."""
    title: str
    body: str


@dataclass
class Notification(BaseNotification):
    """Uses compound type annotations that reference other classes."""
    channel: NotificationChannel | None = None
    priority: Optional[NotificationPriority] = None
    recipients: list[str] = field(default_factory=list)
